examples/structured_2d_dgsem/elixir_advection_basic_coupled.jl

```diff
@@ -0,0 +1,105 @@
+#     │     system #1      │     system #2      │
+#     │     =========      │     =========      │
+#     │                    │                    │
+#     │                    │                    │
+#     │                    │                    │
+#     │                    │                    │
+#     │         coupled -->│<-- coupled         │
+#     │                    │                    │
+#     │<-- coupled         │         coupled -->│
+#     │                    │                    │
+#     │                    │                    │
+#     │    ↓ periodic ↓    │    ↓ periodic ↓    │
+#     └────────────────────┴────────────────────┘
+# (-1, -1)                                   ( 1, -1)
+
+# First mesh is the left half of a [-1,1]^2 square
+coordinates_min1 = (-1.0, -1.0) # minimum coordinates (min(x), min(y))
+coordinates_max1 = ( 0.0,  1.0) # maximum coordinates (max(x), max(y))
+
+cells_per_dimension1 = (8, 16)
+
+mesh1 = StructuredMesh(cells_per_dimension1, coordinates_min1, coordinates_max1)
+
+# A semidiscretization collects data structures and functions for the spatial discretization
+semi1 = SemidiscretizationHyperbolic(mesh1, equations, initial_condition_convergence_test, solver,
+                                     boundary_conditions=(
+                                       # Connect left boundary with right boundary of right mesh
+                                       x_neg=BoundaryConditionCoupled(2, (:end, :i_forward), Float64),
+                                       # Connect right boundary with left boundary of right mesh
+                                       x_pos=BoundaryConditionCoupled(2, (:begin, :i_forward),  Float64),
+                                       y_neg=boundary_condition_periodic,
+                                       y_pos=boundary_condition_periodic))
+
+
+# Second mesh is the right half of a [-1,1]^2 square
+coordinates_min2 = (0.0, -1.0) # minimum coordinates (min(x), min(y))
+coordinates_max2 = (1.0,  1.0) # maximum coordinates (max(x), max(y))
+
+cells_per_dimension2 = (8, 16)
+
+mesh2 = StructuredMesh(cells_per_dimension2, coordinates_min2, coordinates_max2)
+
+semi2 = SemidiscretizationHyperbolic(mesh2, equations, initial_condition_convergence_test, solver,
+                                     boundary_conditions=(
+                                       # Connect left boundary with right boundary of left mesh
+                                       x_neg=BoundaryConditionCoupled(1, (:end, :i_forward), Float64),
+                                       # Connect right boundary with left boundary of left mesh
+                                       x_pos=BoundaryConditionCoupled(1, (:begin, :i_forward),  Float64),
+                                       y_neg=boundary_condition_periodic,
+                                       y_pos=boundary_condition_periodic))
+
+# Create a semidiscretization that bundles semi1 and semi2
+semi = SemidiscretizationCoupled((semi1, semi2))
```

Comment:
A remnant from a previous more complex version with multiple arguments. The first argument was the list of semidiscretizations. Since it is still treated as list and we like to expand on it, I suggest to keep it like this.